protocols/v2/roles-logic-sv2/src/utils.rs

```diff
@@ -172,7 +172,7 @@ fn reduce_path<T: AsRef<[u8]>>(coinbase_id: [u8; 32], path: &[T]) -> [u8; 32] {
 /// [3] https://en.wikipedia.org/wiki/Negative_hypergeometric_distribution
 /// bdiff: 0x00000000ffff0000000000000000000000000000000000000000000000000000
 /// https://en.bitcoin.it/wiki/Difficulty#How_soon_might_I_expect_to_generate_a_block.3F
-pub fn hash_rate_to_target(h: f32, share_per_min: f32) -> U256<'static> {
+fn _hash_rate_to_target(h: f32, share_per_min: f32) -> [u8; 32] {
```

Comment:
not sure that we need `hash_rate_to_target_be`.
Here we should only have the function `hash_rate_to_target` and it should return an `U256`. We should also remove the possibility to do `U256` from `[u8;32]` and `U256` try from `Vec<u8>` 

The issue here is that we treat `U256` like a generic byte array but in the spec is defined as a number.  So the only way to construct an `U256` should be using the methods `from_le_bytes(b: [u8;32]) -> Self` and `from_be_bytes(b: [u8;32]) -> Self`

Reply:
Okay, I can change this. I added this because in the `difficulty_from_target` function in the Tproxy is based on BE targets and in the tproxy difficulty management PR I just kept it that way. Will change back and handle it in the tproxy in the the diff management PR